mungegithub/mungers/milestone-maintainer.go

```diff
@@ -0,0 +1,461 @@
+	// Only target issues
+	if obj.IsPR() {
+		return true
+	}
+
+	// Only target issues with an assigned milestone
+	milestone, ok := obj.ReleaseMilestone()
+	if !ok || len(milestone) == 0 {
+		return true
+	}
+
+	// Only target issues in the active milestone
+	return milestone != m.activeMilestone
+}
+
+// notificationState returns the state required to ensure the munger's
+// notification is kept current.  If a nil return value is returned,
+// no action should be taken.
+func notificationState(obj *github.MungeObject, comment *c.Comment, botName string) (string, *milestoneNotification) {
+	notification := c.ParseNotification(comment)
+
+	kindLabel, priorityLabel, sigLabels, labelErrors := checkLabels(obj.Issue)
+
+	labelsComplete := len(labelErrors) == 0
+	if labelsComplete {
+		return milestoneLabelsCompleteLabel, labelsCompleteState(obj.Issue, notification, kindLabel, priorityLabel, sigLabels)
+	}
+
+	isBlocker := (priorityLabel == priorityCriticalUrgent)
+	var removeIn *time.Duration
+	// Blockers are never removed from the milestone so grace period computation can be skipped
+	if !isBlocker {
+		removeIn = gracePeriodRemaining(obj, botName)
+		if removeIn == nil {
+			return "", nil
+		}
+	}
+	// removeIn is guaranteed to be non-nil for non-blockers
+	if isBlocker || *removeIn >= 0 {
+		var createdAt *time.Time
+		// createdAt should be nil for blockers to avoid repeatedly posting the same warning message
+		if !isBlocker && comment != nil {
+			createdAt = comment.CreatedAt
+		}
+		return milestoneLabelsIncompleteLabel, labelsIncompleteState(obj.Issue, notification, labelErrors, createdAt, removeIn)
+	}
+	return milestoneRemovedLabel, removalState(obj.Issue, notification, labelErrors, *obj.Issue.Milestone.Title)
+}
+
+// latestNotificationComment returns the most recent notification
+// comment posted by the munger.
+//
+// Since the munger is careful to remove existing comments before
+// adding new ones, only a single notification comment should exist.
+func latestNotificationComment(obj *github.MungeObject, botName string) (*c.Comment, bool) {
+	issueComments, ok := obj.ListComments()
+	if !ok {
+		return nil, false
+	}
+	comments := c.FromIssueComments(issueComments)
+	notificationMatcher := c.MungerNotificationName(milestoneNotifierName, botName)
+	notifications := c.FilterComments(comments, notificationMatcher)
+	return notifications.GetLast(), true
+}
+
+// labelsCompleteState returns the notification state for an issue
+// that has the required labels to remain in the active milestone.
+func labelsCompleteState(issue *githubapi.Issue, notification *c.Notification, kindLabel, priorityLabel string, sigLabels []string) *milestoneNotification {
+	notificationIsCurrent := notification != nil && notification.Arguments == milestoneLabelsComplete
+	if notificationIsCurrent {
+		return nil
+	}
+
+	mention := mungerutil.GetIssueUsers(issue).AllUsers().Mention().Join()
+	message := approvers.GenerateTemplateOrFail(`{{.mention}}
```

Comment:
Please fix the indentation for this function call and the other places where `GenerateTemplateOrFail` is called. You might want to declare the template string as a `const` at the beginning of the func since you can't indent a multi-line literal without including the indentations in the literal itself.